src/software/thunderscope/replay/proto_player.py

```diff
@@ -174,6 +235,52 @@ def set_playback_speed(self, speed):
             self.playback_speed = 1.0 / float(speed)
             self.play()
 
+    def single_step_backward(self):
```

Comment:
this implementation isn't single stepping through "ai ticks" or "world frames" (w/e we'd like to call it), its rather single stepping through each proto in the log

The reason why I didn't implement it this way in the original pr is that we are single stepping through the log entries in order, so you would step through the path vis, then the obstacle proto, then a bunch of named value protos, some logs, before the next world which doesn't seem like what we want. It could be like ~20 clicks before the next world proto

I was clicking forward/backward step and nothing was changing on each click

I think the implementation we talked about a while back was stepping back to the previous world proto and then playing the data forward for x amount of time to draw the rest of the data.

Reply:
change naming to step

Reply:
fair enough, we can make an improvement in the future where the single step forward runs until it sees a world proto

or solve single/backward step together differently